modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -219,6 +219,22 @@ Error GDScriptAnalyzer::check_class_member_name_conflict(const GDScriptParser::C
 	return OK;
 }
 
+void GDScriptAnalyzer::get_script_classes(GDScriptParser::ClassNode *p_node, RBSet<GDScriptParser::ClassNode *> &p_set) {
```

Comment:
The name implies you are getting all classes that are defined in scripts, but nothing else. Something about the name `script_classes` makes it a little hard for me to understand what it does. Would it make more sense to say something like `get_context_classes` or `get_outer_and_base_classes` or something more in line with its intent? Those names are terrible too :P

If nothing else, perhaps some comments explaining exactly what the function does would be nice for future readers :)

It also seems like it doesn't really check for native/builtin classes? Should it?

Reply:
I'll name the function `get_class_node_current_scope_classes()`.

Reply:
I love the name!!!! That's perfect.

And just making sure: native/builtin class checks are already made going all the way up outer-class & extends hierarchy? Because maybe we're looking for an enum, say, somewhere in `Object`, and you might need to follow up a few GDScript classes before you get there?